src/Features/Lsif/Generator/CompilerInvocation.cs

```diff
@@ -90,20 +91,25 @@ public static async Task<CompilerInvocation> CreateFromInvocationInfoAsync(Compi
             var projectId = ProjectId.CreateNewId(invocationInfo.ProjectFilePath);
 
             var projectInfo = ProjectInfo.Create(
-                projectId,
-                VersionStamp.Default,
-                name: Path.GetFileNameWithoutExtension(invocationInfo.ProjectFilePath),
-                assemblyName: parsedCommandLine.CompilationName!,
-                language: languageName,
-                filePath: invocationInfo.ProjectFilePath,
-                outputFilePath: parsedCommandLine.OutputFileName,
+                new ProjectInfo.ProjectAttributes(
```

Comment:
Why the use of attributes here? This feels like it's masking over an API wart since that type wouldn't be available publically?

Reply:
`ProjectInfo.Create` doesn't have all parameters that we need to set. Some of them are internal-only. Adding more overloads to `ProjectInfo.Create` is possible but adds a lot of boiler plate and is kind of messy. I think we should have made `ProjectAttributes` public and top level type.